tcms/testcases/models.py

```diff
@@ -57,6 +57,14 @@ class TestCase(TCMSActionModel):
     requirement = models.CharField(max_length=255, blank=True, null=True)
     notes = models.TextField(blank=True, null=True)
     text = models.TextField(blank=True)
+    setup_duration = models.DurationField(db_index=True, null=True, blank=True) or 0
+    testing_duration = models.DurationField(db_index=True, null=True, blank=True) or 0
+
+    @property
+    def expected_duration(self):
+        if not self.setup_duration and not self.testing_duration:
```

Comment:
Condition is `or` because any of the 2 fields may be `None`. Alternatively swap the order of the return statements if you want to use an `and` expression.

Reply:
Furthermore if one of these fields is None but the other one is not we may assume that there's no setup for example and the expected duration will be equal to the value of the other field alone.